src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -446,6 +446,69 @@ class Direct$Type$Buffer$RW$$BO$
 
     // --- Methods to support CharSequence ---
 
+    private static final int APPEND_BUF_SIZE = 1024;
+
+    private $Type$Buffer appendChars(CharSequence csq, int start, int end) {
+        checkSession();
```

Comment:
This template is for direct buffers so I'm not 100% sure that checkSession is needed here. If the session is closed then SMA.copyMemory used in putArray will throw.

Reply:
This is probably to avoid allocating the temporary `char[] buf` array when the session is known to be closed.

Reply:
The primary direct buffer constructor is
```
    // Primary constructor
    //
    Direct$Type$Buffer$RW$(int cap) {                   // package-private
```
so in general the `segment` will be `null` and checking it provides no value.

Reply:
> This template is for direct buffers so I'm not 100% sure that checkSession is needed here.

Commit 739c33a removes `checkSession()`.